hudi-common/src/main/java/org/apache/hudi/common/fs/inline/InLineFSUtils.java

```diff
@@ -54,21 +57,26 @@ public static Path getInlineFilePath(Path outerPath, String origScheme, long inL
   }
 
   /**
-   * Inline file format
-   * "inlinefs://<path_to_outer_file>/<outer_file_scheme>/?start_offset=start_offset>&length=<length>"
-   * Outer File format
-   * "<outer_file_scheme>://<path_to_outer_file>"
+   * InlineFS Path format:
+   * "inlinefs://path/to/outer/file/outer_file_schema/?start_offset=start_offset>&length=<length>"
    * <p>
-   * Eg input : "inlinefs://file1/sa3/?start_offset=20&length=40".
-   * Output : "sa3://file1"
+   * Outer File Path format:
+   * "outer_file_schema://path/to/outer/file"
+   * <p>
+   * Example
+   * Input: "inlinefs://file1/s3a/?start_offset=20&length=40".
```

Comment:
I am thinking if we should encode the no of slashes as well when inline path is generated and then later re-use when we regenerate the original path later? so that it works for any schemes.
```
inlinefs://file1/s3a/?start_offset=20&length=40&slashes=2
```

Reply:
I have a different thought on this. The existing inlinefs paths doesn't follow the right URI format. URIs have the format `<schema>://path/to/file/?options=args&..`. So, instead of encoding  number of slashes, we need to make all encoded/converted paths to comply to standard format which includes the schema separator `://`. But doing so will change a quite a few of existing inlinefs path handling and maybe not the right idea to do late in the release. I can file a new ticket to take it up for .11 to make inlinefs paths better. your thoughts please?

Reply:
I agree with you. lets see if we can make that fix. most changes should be confined to InLineFSUtils. Can you take a stab. We don't have inline FS in last release. so, wanted to get it in good shape with 0.10.0 atleast the format. do not want to make changes in this regard later.

Reply:
Also, did you think about how to make this work for [all storage schemes](https://github.com/apache/hudi/blob/master/hudi-common/src/main/java/org/apache/hudi/common/fs/StorageSchemes.java)?

Reply:
I feel, we can get the right fix in for all storage schemes as per your suggestion. lets sync up on this.

Reply:
Right, made all non LocalFileSystem follow the same URI format of `scheme://path/to/file`